src/api/index.tsx

```diff
@@ -364,26 +435,52 @@ export const createOrUpdateComment = async ({
   });
 };
 
-export type GetCommentsResponse = PaginatedResponse<ExternalComment[]>;
+type GetCommentsResultsItem = ExternalComment;
+
+export type GetCommentsResponse = PaginatedResponse<GetCommentsResultsItem[]>;
+
+export type GetCommentsParams = {
+  _callApi?: typeof callApi;
+  addonId: number;
+  apiState: ApiState;
+  endpointUrl?: GetCommentsResponse['next'];
+  versionId: number;
+};
 
 export const getComments = async ({
   _callApi = callApi,
   addonId,
   apiState,
+  endpointUrl,
   versionId,
-}: {
-  _callApi?: typeof callApi;
-  addonId: number;
-  apiState: ApiState;
-  versionId: number;
-}) => {
+}: GetCommentsParams) => {
+  let endpoint;
+
+  if (!endpointUrl) {
+    endpoint = `reviewers/addon/${addonId}/versions/${versionId}/draft_comments`;
+  }
+
   return _callApi<ResponseOnly<GetCommentsResponse>>({
     apiState,
-    endpoint: `reviewers/addon/${addonId}/versions/${versionId}/draft_comments`,
+    endpointUrl: endpointUrl || undefined,
```

Comment:
I'm having a bit of trouble understanding why `getComments` needs to have special handling for `endpointUrl`, and why we cannnot just handle that in `callApi`. Couldn't `callApi` decide what to do based on what it receives for both `endpoint` and `endpointUrl`? I.e., if there is an `endpointUrl` use that, otherwise use `endpoint`?

Reply:
> I'm having a bit of trouble understanding why `getComments` needs to have special handling for `endpointUrl`, and why we cannnot just handle that in `callApi`. Couldn't `callApi` decide what to do based on what it receives for both `endpoint` and `endpointUrl`? I.e., if there is an `endpointUrl` use that, otherwise use `endpoint`?

`callApi()` could make that decision but how would we know that it's the right decision? It could be wrong and that would introduce subtle bugs. The caller should be the one to decide.

Just as an exercise, I committed https://github.com/mozilla/addons-code-manager/pull/1190/commits/fb15d6004eb812577fcc85426f3e9fd595ec8642 to introduce the behavior you described but it feels more risky and I don't think it's helpful. It allowed me to remove an if statement from `getComments()` but overall the diff adds more code than it takes away.